.github/workflows/icon-review.yml

```diff
@@ -3,107 +3,123 @@ name: 🏞️ Icon Review
 on:
   pull_request_target:
     paths:
+      # Only trigger the workflow if SVG files in the icons directory are changed
       - "icons/*.svg"
 
+# Permissions for the workflow
 permissions:
   contents: read
+  pull-requests: write
+  issues: write
 
 jobs:
   icon-review:
     name: Icon Review
     runs-on: ubuntu-latest
-    env:
-      TARGET_BRANCH: ${{ github.event.pull_request.base.ref }}
-
-    permissions:
-      pull-requests: write
 
     steps:
-      - name: 📥 Checkout Fork
-        uses: actions/checkout@11bd71901bbe5b1630ceea73d27597364c9af683 # v4.2.2
-        with:
-          ref: ${{ github.event.pull_request.head.ref }}
-          repository: ${{ github.event.pull_request.head.repo.full_name }}
-          fetch-depth: 0
-          path: fork
-          persist-credentials: false
-
-      - name: 📥 Checkout Original
+      - name: 📥 Checkout
         uses: actions/checkout@11bd71901bbe5b1630ceea73d27597364c9af683 # v4.2.2
         with:
           fetch-depth: 0
-          path: main
           persist-credentials: false
 
       - name: 🔧 Setup Bun
         uses: oven-sh/setup-bun@4bc047ad259df6fc24a6c9b0f9a0cb08cf17fbe5 # v2
+
+        with:
+          bun-version-file: ".bun-version"
+
+      - name: 🔍 Get changed SVG files
+        id: get_changed_files
+        uses: actions/github-script@v7.0.1
         with:
-          bun-version-file: "main/.bun-version"
+          script: |
+            const { data: files } = await github.rest.pulls.listFiles({
+              owner: context.repo.owner,
+              repo: context.repo.repo,
+              pull_number: context.issue.number,
+            });
+
+            const svgFiles = files
+              .filter(file => file.filename.startsWith('icons/') && file.filename.endsWith('.svg'))
+              .map(file => file.filename);
+
+            const changedSvgFilesCount = svgFiles.length;
+
+            core.setOutput('changed_svg_files', svgFiles.join(' '));
+            core.setOutput('changed_svg_files_count', changedSvgFilesCount);
 
       - name: 🔍 Review SVG files
-        working-directory: main
         run: |
-          files=$(git diff --no-index ../main ../fork --diff-filter=ACMRTUX --name-only | grep '^\.\./fork/icons/.*\.svg$')
-          filesCount=$(echo "$files" | wc -l)
-          bunx --bun svg-icon-review@2.1.0 --bigIcon ${files}
-          echo svg_files_count=$filesCount >> $GITHUB_ENV
+          bunx svg-icon-review@2.1.0 --bigIcon ${{ steps.get_changed_files.outputs.changed_svg_files }}
```

Comment:
I still can't make it work. I just tried the latest changes of this PR on my fork and it's not using the updated SVG file from the PR but from the main branch.

The old workflow uses this HTML to render the image:
```html
<span class="icon-preview" style="background-image: url('/home/runner/work/vscode-material-icon-theme-labs/vscode-material-icon-theme-labs/fork/icons/angular.svg')"></span>
```

The changed workflow of this PR uses this HTML to render the image:

```html
<span class="icon-preview" style="background-image: url('/home/runner/work/vscode-material-icon-theme-labs/vscode-material-icon-theme-labs/icons/angular.svg')"></span>
```

And it's producing the image for the wrong icon. 

Maybe I'm doing something wrong, but I feel like this approach is not working properly.

--- 

I did the following steps:

1. Fork the repo vscode-material-icon-theme
2. Copy and paste the changes from this PR to the main branch of the fork
3. Create a new branch in the fork (with the changed workflow on main)
4. Change some icons
5. Commit and push changes
6. Create a new PR of this branch in the fork which targets to main in the fork
7. Let the workflows run

For instance, I've changed the angular.svg file to have a green color: 
![image](https://github.com/user-attachments/assets/a3dc0ab6-118b-4319-a942-05a31614bc9f)

The preview image which is shown looks like this:
![image](https://github.com/user-attachments/assets/167c84c3-a6ee-4698-bb6c-ceb028ee31d3)

Reply:
> I still can't make it work. I just tried the latest changes of this PR on my fork and it's not using the updated SVG file from the PR but from the main branch.
> 
> The old workflow uses this HTML to render the image:
> 
> ```
> <span class="icon-preview" style="background-image: url('/home/runner/work/vscode-material-icon-theme-labs/vscode-material-icon-theme-labs/fork/icons/angular.svg')"></span>
> ```
> 
> The changed workflow of this PR uses this HTML to render the image:
> 
> ```
> <span class="icon-preview" style="background-image: url('/home/runner/work/vscode-material-icon-theme-labs/vscode-material-icon-theme-labs/icons/angular.svg')"></span>
> ```
> 
> And it's producing the image for the wrong icon.
> 
> Maybe I'm doing something wrong, but I feel like this approach is not working properly.
> 
> I did the following steps:
> 
> 1. Fork the repo vscode-material-icon-theme
> 2. Copy and paste the changes from this PR to the main branch of the fork
> 3. Create a new branch in the fork (with the changed workflow on main)
> 4. Change some icons
> 5. Commit and push changes
> 6. Create a new PR of this branch in the fork which targets to main in the fork
> 7. Let the workflows run
> 
> For instance, I've changed the angular.svg file to have a green color: ![image](https://private-user-images.githubusercontent.com/12248527/383679677-a3dc0ab6-118b-4319-a942-05a31614bc9f.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MzA5MTgzMTMsIm5iZiI6MTczMDkxODAxMywicGF0aCI6Ii8xMjI0ODUyNy8zODM2Nzk2NzctYTNkYzBhYjYtMTE4Yi00MzE5LWE5NDItMDVhMzE2MTRiYzlmLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDExMDYlMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQxMTA2VDE4MzMzM1omWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTEwOTFlYzkxNzg2NmI2MTRjNWM0NWY5MDRlNGE5OGNiNTczNTQxOTBlYWRjN2QxMjk5OWJlNmNiNWZlNTdmNTAmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0In0.tOKwYlkQizbi-_It7iGhjhAQ0lo5FlzDpUesIMh9_4g)
> 
> The preview image which is shown looks like this: ![image](https://private-user-images.githubusercontent.com/12248527/383679609-167c84c3-a6ee-4698-bb6c-ceb028ee31d3.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MzA5MTgzMTMsIm5iZiI6MTczMDkxODAxMywicGF0aCI6Ii8xMjI0ODUyNy8zODM2Nzk2MDktMTY3Yzg0YzMtYTZlZS00Njk4LWJiNmMtY2ViMDI4ZWUzMWQzLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDExMDYlMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQxMTA2VDE4MzMzM1omWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTE0ZGYwYmZmZTNmMDI3ZjVkZDlkMDkzZmExN2VhMjJiNjBhMjIyZGZlMzFmMGQ0MmYyOWExYmEyN2U1YWUwOGImWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0In0.088xasKZtIcqmu798-pYmwY-iKWl4LS-R7J_-gXzN90)

I have an idea

Let's test it in production

Reply:
"Let's test it in production" you mean merging it to main?

Reply:
> "Let's test it in production" you mean merging it to main?

Yes 😈👹

Reply:
Ok let's try it...